applications/vanilla/models/class.categorymodel.php

```diff
@@ -2801,4 +2805,126 @@ public function recalculateTree() {
     public static function flattenTree($categories) {
         return self::instance()->collection->flattenTree($categories);
     }
+
+    /**
+     * Adjust the aggregate post counts for a category.
+     *
+     * @param int|string|object|array $category
+     * @param int $offset
+     */
+    private static function adjustAggregateCounts($category, $offset) {
+        $categoryID = false;
+        $offset = intval($offset);
+
+        if (is_numeric($category)) {
+            $categoryID = $category;
+        } elseif (is_string($category)) {
+            $category = self::instance()->collection->get($category);
+        }
+
+        if (is_array($category) || is_object($category)) {
+            $categoryID = val('CategoryID', $category);
+        }
+
+        if (empty($categoryID)) {
+            return;
+        }
```

Comment:
I guess the real question is: Wherever `decrementAggregateCounts` and `incrementAggregateCounts` are used, do we really not have a reliable data type available? If not, is that perhaps the problem we should be solving as part of this refactor rather than making these helpers so permissive?